lib/fresh_connection/connection_factory.rb

```diff
@@ -22,19 +25,66 @@ def spec
       @spec ||= build_spec
     end
 
+    # when building a spec from envars, there may be no database.yml file, so
+    # be careful to avoid implicit dependency on it or derived contents.
+
     def build_spec
-      config = ar_spec.config.symbolize_keys
-      group_config = config[@group]
+      url = group_url
+      if url
+        build_and_set_group_spec_and_url(url)
+      else
+        build_spec_from_config
+      end
+    end
+
+    # look for an envar with the current @group name, if it's not found, and if
+    # the group is :replica, then look for an envar using :replica1, which if
+    # found becomes the new @group
 
-      # provide backward compatibility for older :slave usage
-      if !group_config && @group == :replica && config.key?(:slave)
-        ActiveSupport::Deprecation.warn(
-          "'slave' in database.yml is deprecated and will ignored from version 2.4.0. use 'replica' insted."
-        )
-        group_config = config[:slave]
+    def group_url
+      url = database_group_url(@group.to_s)
+      if url.nil? && @group == :replica
+        url = database_group_url('replica1')
+        @group = :replica1 if url
```

Comment:
The idea is that for code that specifies `:replica`, which is the default group name, but the envars `DATABASE_REPLICA1_URL` and `DATABASE_REPLICA2_URL` are defined, then the _effective_ group name is `:replica1`, since that correlates to the envar that was actually defined.